File: apps/relay-go/pkg/types/settlement.go
package types

// SettlementOutcome enumerates the resolution classes for one
// In-Between round.
type SettlementOutcome string

const (
	OutcomeWin                 SettlementOutcome = "win"
	OutcomeLoss                SettlementOutcome = "loss"
	OutcomePass                SettlementOutcome = "pass"
	OutcomeConsecutivePenalty  SettlementOutcome = "consecutive_penalty"
	OutcomeEqualPenalty        SettlementOutcome = "equal_penalty"
)

// SettlementResult is the result of a Settle action.
type SettlementResult struct {
	RoundStateHash      Hash256           `json:"round_state_hash"`
	Outcome             SettlementOutcome `json:"outcome"`
	BetAmount           Satoshis          `json:"bet_amount"`
	ActingPlayerSeat    Seat              `json:"acting_player_seat"`
	AmountWonOrLost     Satoshis          `json:"amount_won_or_lost"`
	ResultingPotValue   Satoshis          `json:"resulting_pot_value"`
	ResultingBalances   []Satoshis        `json:"resulting_balances"`
}

// RecoveryRecord captures one global recovery outcome.
type RecoveryRecord struct {
	StalledStateHash    Hash256                     `json:"stalled_state_hash"`
	RecoveryTrigger     string                      `json:"recovery_trigger"`
	FinalDistribution   []RecoveryDistributionEntry `json:"final_distribution"`
	PenaltyBurntSats    Satoshis                    `json:"penalty_burnt_sats"`
}

// RecoveryDistributionEntry is one party's row in the recovery payout.
type RecoveryDistributionEntry struct {
	Seat                 Seat     `json:"seat"`
	RefundSats           Satoshis `json:"refund_sats"`
	PenaltyPaidSats      Satoshis `json:"penalty_paid_sats"`
	PenaltyReceivedSats  Satoshis `json:"penalty_received_sats"`
}

File: packages/protocol-types/src/settlement.ts
/**
 * Settlement and recovery records.
 */

import type { Hash256, Satoshis, Seat } from './primitives.js';

/** Outcome classes for one round in In-Between. */
export type SettlementOutcome =
  | 'win'
  | 'loss'
  | 'pass'
  | 'consecutive_penalty'
  | 'equal_penalty';

/** Result of a {@link SettleAction}. */
export interface SettlementResult {
  readonly round_state_hash: Hash256;
  readonly outcome: SettlementOutcome;
  readonly bet_amount: Satoshis;
  readonly acting_player_seat: Seat;
  readonly amount_won_or_lost: Satoshis;
  readonly resulting_pot_value: Satoshis;
  /** Per-seat balances after this settlement. */
  readonly resulting_balances: readonly Satoshis[];
}

/** Record of a recovery action's effect on the session. */
export interface RecoveryRecord {
  readonly stalled_state_hash: Hash256;
  readonly recovery_trigger: string;
  readonly final_distribution: readonly {
    readonly seat: Seat;
    readonly refund_sats: Satoshis;
    readonly penalty_paid_sats: Satoshis;
    readonly penalty_received_sats: Satoshis;
  }[];
  readonly penalty_burnt_sats: Satoshis;
}
